### PR TITLE
Improve documentation of Arpack

### PR DESCRIPTION
**Pull request by Rombur**
The first commit remove doxygen marker in some comments. They would show up in our documentation when they shouldn't see the end of documentation of [solve](http://dealii.org/developer/doxygen/deal.II/classArpackSolver.html#ae49067a1359b8d6794b92e1d0b405df6). Also improve documentation of `AdditionalData`

### CODE REVIEW

File: include/deal.II/lac/arpack_solver.h [hunk 1]
```diff
@@ -160,18 +160,36 @@ class ArpackSolver : public Subscriptor
...
+     * eigenvalues of largest magnitude for a non-symmetric problem).
+     */
+    explicit AdditionalData(
       const unsigned int number_of_arnoldi_vectors = 15,
       const WhichEigenvalues eigenvalue_of_interest = largest_magnitude,
       const bool symmetric = false);
+
+    /**
+     * Number of Arnoldi/Lanczos vectors. This number should be less than the
+     * size of the problem but greater than 2 times the number of eigenvalues
```
Comment (davydden) [inline include/deal.II/lac/arpack_solver.h]: i think it is recommended to be around that, but strictly speaking it does not have to.

Comment (Rombur) [reply include/deal.II/lac/arpack_solver.h]: I just looked at the code and we have an assert that checks for that value see [here](https://github.com/dealii/dealii/pull/5553/files/f6d6897b5d5873e89813632811cceddd1797c842..e82eed83b10a36fa80ce390e3e52da9f44f2feb6#diff-8fb043a1e58ccb9048d7210b151893a0R472). I have no idea if that constraint makes sense or not.

Comment (davydden) [reply include/deal.II/lac/arpack_solver.h]: interesting... keep it as is then

**Comment by davydden** [comment]
./run-tests

**Pull request merged**